core/jobs/batch_jobs/voiceover_migration_job.py

```diff
@@ -0,0 +1,358 @@
+
+    @classmethod
+    def update_entity_voiceover_for_given_id(
+        cls,
+        entity_type: str,
+        entity_id: str,
+        entity_version: int,
+        accent_code: str,
+        content_id: str,
+        voiceover_dict: state_domain.VoiceoverDict,
+        entity_voiceover_id_to_entity_voiceovers: Dict[
+            str, voiceover_models.EntityVoiceoversModel]
+    ) -> Dict[str, voiceover_models.EntityVoiceoversModel]:
+        """Updates the entity voiceover model instance for the given language
+        accent code for a specific content of an exploration.
+
+        Args:
+            entity_type: str. The entity type of an entity voiceovers instance.
+            entity_id: str. The given entity ID.
+            entity_version: int. The version number of the given exploration.
+            accent_code: str. The language accent code for the given entity
+                voiceover instance.
+            content_id: str. The content ID for a specific content in an
+                exploration.
+            voiceover_dict: VoiceoverDict. A voiceover dict that should be
+                added to the entity voiceover instance of the given language
+                accent code of the content.
+            entity_voiceover_id_to_entity_voiceovers:
+                dict(str, EntityVoiceoversModel). A dict with entity voiceover
+                ID as keys and its corresponding EntityVoiceoversModel as
+                values.
+
+        Returns:
+            dict(str, EntityVoiceoversModel). The updated dict with entity
+            voiceover ID as keys and its corresponding EntityVoiceoversModel as
+            values.
+        """
+        entity_voiceover_id = (
+            voiceover_models.EntityVoiceoversModel.generate_id(
+                entity_type,
+                entity_id,
+                entity_version,
+                accent_code
+            )
+        )
+
+        if entity_voiceover_id not in entity_voiceover_id_to_entity_voiceovers:
+            entity_voiceovers_object = (
+                cls.create_entity_voiceovers_model_instance(
+                    entity_id, entity_type, entity_version, accent_code))
+        else:
+            entity_voiceovers_object = (
+                entity_voiceover_id_to_entity_voiceovers[entity_voiceover_id])
+
+        entity_voiceovers_object.voiceovers[content_id] = {
+            'manual': voiceover_dict
+        }
+
+        entity_voiceover_id_to_entity_voiceovers[entity_voiceover_id] = (
+            entity_voiceovers_object)
+        return entity_voiceover_id_to_entity_voiceovers
+
+    @classmethod
+    def generate_entity_voiceover_models(
+        cls,
+        element: ExplorationAndVoiceArtistLinkType,
+        voice_artist_metadata_models_list: List[
+            voiceover_models.VoiceArtistMetadataModel]
+    ) -> List[voiceover_models.EntityVoiceoversModel]:
+        """Generates an entity voiceovers model using the voiceover data stored
```

Comment:
Confusing --- this says "model", the method name says "models", and the docstring is inconsistent because it says a voiceovers model (singular) is generated from exploration models (plural), so I don't know what is going on.